src/snowcli/cli/nativeapp/run_processor.py

```diff
@@ -0,0 +1,214 @@
+from pathlib import Path
+from textwrap import dedent
+from typing import Dict
+
+import jinja2
+from rich import print
+from snowcli.cli.nativeapp.constants import (
+    COMMENT_COL,
+    INTERNAL_DISTRIBUTION,
+    LOOSE_FILES_MAGIC_VERSION,
+    SPECIAL_COMMENT,
+    VERSION_COL,
+)
+from snowcli.cli.nativeapp.exceptions import (
+    ApplicationAlreadyExistsError,
+    ApplicationPackageAlreadyExistsError,
+    InvalidPackageScriptError,
+    MissingPackageScriptError,
+)
+from snowcli.cli.nativeapp.manager import (
+    NativeAppCommandProcessor,
+    NativeAppManager,
+    ensure_correct_owner,
+    generic_sql_error_handler,
+)
+from snowcli.cli.object.stage.diff import DiffResult
+from snowcli.cli.object.stage.manager import StageManager
+from snowflake.connector import ProgrammingError
+
+
+class NativeAppRunProcessor(NativeAppManager, NativeAppCommandProcessor):
+    def __init__(self, project_definition: Dict, project_root: Path):
+        super().__init__(project_definition, project_root)
+
+    def create_app_package(self) -> None:
+        """
+        Creates the application package with our up-to-date stage if none exists.
+        """
+
+        # 1. Check for existing existing application package
+        show_obj_row = self.get_existing_app_pkg_info()
+
+        if show_obj_row:
+            # 1. Check for the right owner role
+            ensure_correct_owner(
+                row=show_obj_row, role=self.package_role, obj_name=self.package_name
+            )
+
+            # 2. Check distribution of the existing app package
+            actual_distribution = self.get_app_pkg_distribution_in_snowflake
+            if not self.is_app_pkg_distribution_same_in_sf():
+                print(
```

Comment:
It will break output when using `--format=JSON`. But I understand requirement of progress messages and as our task for such messages (SNOW-986249) is still in "to do", I think it's ok for now to use just a `print` here.

However we should revisit these prints after implementation of SNOW-986249 to make JSON output working correctly for NADE commands.

Reply:
Agreed, this a temporary solution that we plan to revisit once the above mentioned ticket is picked up.